### ISSUE TITLE
Cleanup t.Logf & t.Fail

### ISSUE THREAD

**Issue by miekg** [miekg/dns#314]
Tests have a lot of:
```
		t.Logf("error not ErrTruncated from header-only unpack: %v", err)
		t.Fail()
```
This should just be compacted into `t.Fatalf(...)`

**Comment by andrewtj** [comment]
I'm pretty sure I've seen you comment before that some (all?) of those are intentional because `t.Fatalf()` is `t.Logf()` followed by `t.FailNow()` (as opposed to `t.Fail()`).

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [dns] Cleanup t.Logf & t.Fail (..." ]
>I'm pretty sure I've seen you comment before that some (all?) of those are intentional because `t.Fatalf()` is `t.Logf()` followed by `t.FailNow()` (as opposed to `t.Fail()`).

Yeah, I know :) But its two lines, we can get away by just trimming this to one
line.

/Miek

--
Miek Gieben

**Comment by miekg** [comment]
Closed with #367

**Comment by asergeyev** [comment]
Just note btw (might be useful for future)  t.Logf only prints when test ran with verbosity. Fatalf prints always. Also in places where t.Fail replaced with t.Errorf it'll be "print and continue" vs previous "print and return"
